Automattic/wp-calypso
# HelpCenter: upgrade all endpoints to proxy through Jetpack

## Summary

*I pinged a lot of people for review just to let everyone who has been a part of this journey know*.

This is the @Automattic/vertex **GRAND** HelpCenter fix. For more context start you can read from here - pdDR7T-8r-p2#comment-68. The basic gist is that API calls to the public-api.wordpress.com are not permitted outside of WPCOM this includes Atomic sites because they are not hosted on WPCOM. Because of this, we were not able to search support guides, fetch support guide content, or check support eligibility from Atomic. The only way around this was to proxy our requests from Atomic through the Jetpack Client using `wpcom_json_api_request_as_user` (read more about that here - pb5gDS-wB-p2).

In order to make this switch we needed to create our own copies of the search elements and connect them to our own API calls to new endpoints. That is what this PR handles. We created three endpoints that are registered in our plugin and then use those to either proxy the requests through Jetpack for Atomic or use the appropriate library on WPCOM for Simple sites. As for right now, this is still existing inside ETK, but it will likely be moved to WPCOMSH or Jetpack in the near future.

All new API endpoints are live on WPCOM and can be examined here 
D83399-code
D83757-code
D83571-code

## What to expect

Everything should function as it did before with one change. We chose to return the post_content in the search results for the support guides. This makes the initial search slightly slower but makes the article load instant. It saves on API calls as well.

| Simple | Atomic |
| - | - |
| ![Screen Capture on 2022-07-07 at 02-16-59](https://user-images.githubusercontent.com/33258733/177850712-d9514777-f724-4530-af9a-935b5a517cef.gif) | ![Screen Capture on 2022-07-07 at 02-17-37](https://user-images.githubusercontent.com/33258733/177850653-6ec05cfb-9c91-44f1-96b3-c589b3aef2fb.gif) |

## Testing

1. Clear your sandbox and ensure you are sandboxing the public-api.
2. Pull branch and run `cd apps/editing-toolkit && yarn dev --sync`
3. Go to a sandboxed Simple site and check the HelpCenter is functional
4. Download the [ETK plugin](https://teamcity.a8c.com/repository/download/calypso_WPComPlugins_EditorToolKit/8187678:id/editing-toolkit.zip) and upload it to an AT site
5. Test the same way. The only variation should be that the "Still need help?" button redirects rather than opening in the HelpCenter
